CondFormats/HcalObjects/interface/HcalRecoParamWithPulseShapeT.h

```diff
@@ -0,0 +1,50 @@
+#ifndef CondFormats_HcalObjects_interface_HcalRecoParamWithPulseShapeHostT_h
+#define CondFormats_HcalObjects_interface_HcalRecoParamWithPulseShapeHostT_h
+
+#include "CondFormats/HcalObjects/interface/HcalRecoParamWithPulseShapeSoA.h"
+#include "DataFormats/Portable/interface/PortableCollection.h"
+#include "DataFormats/Portable/interface/PortableHostCollection.h"
+
+template <typename TDev>
+class HcalRecoParamWithPulseShapeT {
+public:
+  using RecoParamCollection = PortableCollection<HcalRecoParamSoA, TDev>;
+  using PulseShapeCollection = PortableCollection<HcalPulseShapeSoA, TDev>;
+
+  class ConstView {
+  public:
+    using RecoParamConstView = typename RecoParamCollection::ConstView;
+    using PulseShapeConstView = typename PulseShapeCollection::ConstView;
+    constexpr ConstView(RecoParamConstView recoView, PulseShapeConstView psView)
+        : recoParamView_{recoView}, pulseShapeView_{psView} {};
+    constexpr float pulseShape(int id) const { return 0; }
```

Comment:
What is this method for ?

Reply:
IIRC our idea was this pattern to allow use along
```cpp
auto const& acc25nsVec = recoParamsWithPS.acc25nsVec(id);
```
instead of the present
```cpp
auto const did = DetId{id};
auto const hashedId = did.subdetId() == HcalBarrel ? did2linearIndexHB(...) : did2linearIndexHE(...);
auto const recoPulseShapeId = recoParamView.ids()[hashedId];
auto const& acc25nsVec = pulseShapeView.acc25nsVec()[recoPulseShapeId];
```
(i.e. abstract the indexing of the `recoParamView.ids()` -> `pulseShapeView` indirection from the calling code)

Looking at this again, as far as I can tell all the six `acc25nsVec` etc. arrays are used only for `compute_pulse_shape_value()` (that is called many times), and that we probably don't want to repeat the `did2linearIndexHB(...)` / `did2linearIndexHE(...)`, how about
* having a member function in this `ConstView()` would take the `id` parameter that would
   * do the calculation from `id` to `hashedId`, and
   * return another View-like object, that would have
      * pointers (or `const` references) to the arrays corresponding to the given `hashedId` as data members, and
      * a member function to do what `compute_pulse_shape_value()` does presently

?

Reply:
how about this?
[4bcedbf](https://github.com/cms-sw/cmssw/pull/44910/commits/4bcedbfa3762f46753bcb6564102898708f5562b)

`hashedId` is used for other conditions as well (not just `recoparamWithPS`), so I'm not including the `id` to `hashedId` computation in this class.

Reply:
I'm not sure I understand how the method
```c++
constexpr float pulseShape(int id) const { return 0; }
```
helps with any of that ?

Reply:
`pulseShape(int id)` was a place-holder function, so I have removed it in the new commit.